### PR TITLE
fix alert banner issues

### LINKED ISSUE #19506: "Testnet mode enabled" banner issues
# Bug Report
followup https://github.com/status-im/status-mobile/pull/19348

## Problem 
[comment]: # (A problem description)

#### Expected behavior
Send button is present as designed
Screenshot 2024-04-03 at 18 23 06

#### Actual behavior
Screenshot 2024-04-03 at 18 21 39

### Reproduction

1. Install app
2. Make sure you have min 1 contact
3. Open chat with your contact
4. Try to send any message in this chat

### Additional Information

- Status version: PR https://github.com/status-im/status-mobile/pull/19348
- Operating System: iOS
- iPhone 13 17.3.1

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/19506

### Testing Note:

- Communities screen, chat screen and create wallet issues should be fixed
- Legacy screens use a native topbar whose position can't be changed. So we can't fix these screens. But these screens are legacy and will be removed.
- Please let me know if there any other screen where alert banner is causing problem

status:  ready

### CODE REVIEW

File: src/status_im/config.cljs [hunk 1]
```diff
@@ -133,3 +133,7 @@
 (def community-accounts-selection-enabled? true)
 (def fetch-messages-enabled? (enabled? (get-config :FETCH_MESSAGES_ENABLED "1")))
 (def test-networks-enabled? (enabled? (get-config :TEST_NETWORKS_ENABLED "0")))
+
+;; Alert banners are disabled for debug builds because alert banners overlay
+;; interfere with react-native debug tools, such as inspector and Perf monitor
```
Comment (ibrkhalil) [inline src/status_im/config.cljs:141]: +1
Makes total sense to disable it on debug builds

**Review by ibrkhalil — APPROVED** [review]

**Comment by pavloburykh** [comment]
@Parveshdhull thank you for the PR. 

Please, take a look at the following issues. All of them are IOS related. Tested on iPhone X, IOS 16.7.7

### ISSUE 1 Composer is overlayed by opened keyboard (IOS)

https://github.com/status-im/status-mobile/assets/97245802/f5526f11-1a42-40b9-8b67-65d707aceca2

**Comment by pavloburykh** [comment]
### ISSUE 2 Save button is overlayed by opened keyboard on Bio/Name screens

**Steps:** 

1. Go to Profile - Edit Profile - Name/Bio
2. Pay attention at Save button

**Actual result:** 

https://github.com/status-im/status-mobile/assets/97245802/2a130147-5246-4679-b99f-8df8ec58ff8a

**Comment by pavloburykh** [comment]
### ISSUE 3 Some legacy settings are overlayed by banner

https://github.com/status-im/status-mobile/assets/97245802/36121cd6-13ee-48e7-bd55-a8225dd5983e

**Comment by pavloburykh** [comment]
Unfortunately I was not able to test this [issue](https://github.com/status-im/status-mobile/issues/19506#issuecomment-2036805742) on IOS, because fetching communities is blocked by https://github.com/status-im/status-mobile/issues/19736

**Comment by Parveshdhull** [comment]
Thank you @pavloburykh for testing the PR and finding these issues. Sorry I only tested android and missed these issues in IOS (for some reason my iphone takes around an hour to install PR builds)

I will push fix for issue 1 and 2 soon. Issue 3 is expected(please check PR description) and overlapping of legacy screens can't be fixed, but I think we can get rid of black area. (will check that)

**Comment by pavloburykh** [comment]
### ISSUE 4 Open keyboard overlays Confirm button on Edit Account Screen (IOS)

**Steps:** 

1. Go to Wallet tab - enter existing account - top right menu button - edit account
2. See open keyboard overlays confirm button 


https://github.com/status-im/status-mobile/assets/97245802/8566e5c2-32e5-48fe-becd-ca72abe25a7a

**Comment by pavloburykh** [comment]
> I will push fix for issue 1 and 2 soon.

Thank you @Parveshdhull!

> Issue 3 is expected(please check PR description)

Sorry @Parveshdhull I have missed this comment. 


>  but I think we can get rid of black area. (will check that)

I think it should be fixed in https://github.com/status-im/status-mobile/pull/19735 I will check after you rebase this PR after fixes to make sure.

**Comment by Parveshdhull** [comment]
@pavloburykh Thank you for finding these, issue 1, 2 and 4 should be fixed now.
(upd: oops component-tests failed, pushing fix)
upd2: fixed

**Comment by pavloburykh** [comment]
> @pavloburykh Thank you for finding these, issue 1, 2 and 4 should be fixed now. (upd: oops component-tests failed, pushing fix) upd2: fixed

@Parveshdhull thanks for the fixes. 

### [ISSUE 2](https://github.com/status-im/status-mobile/pull/19773#issuecomment-2075145811) Save button is overlayed by opened keyboard on Bio/Name screens

is fixed partially. Edit name screen looks fine but on Bio screen keyboard still overlays Save button on IOS. Please take a look. 


https://github.com/status-im/status-mobile/assets/97245802/a806b393-20c4-4000-917e-676047fc3681

**Comment by pavloburykh** [comment]
### ISSUE 5 Expanded composer is overlapped by Banner

Reproducible both on IOS and Android

**Steps:** 

1. Expand the composer
2. See if composer is not overlapped by Banner. 

**Actual result:** expanded composer is overlapped by Banner. Unable to fold the composer by swipe. 


https://github.com/status-im/status-mobile/assets/97245802/7f1ead75-bde0-4c5d-94b3-0f3d8acc50b3

**Comment by Parveshdhull** [comment]
Thank you @pavloburykh for finding these issues, both issues should be fixed now.

**Comment by pavloburykh** [comment]
@Parveshdhull thank you for fixing those issue. Please take a look at another one


### ISSUE 6 Expanded composer is hardly foldable on IOS

Reproducible on IOS only. 

I am hardly able to fold expanded composer. Typically it should work by pulling the edge of the composer. 

Please, take a look how it behaves now:

https://github.com/status-im/status-mobile/assets/97245802/b3db512b-1d7a-4700-942b-b8779b40ded9

**Comment by Parveshdhull** [comment]
**Issue 7**
IOS native image picker cance/select button gets overlapped by alert banner.

**Comment by Parveshdhull** [comment]
Thank you @pavloburykh for finding issue 6, it was an interesting one. Should be fixed now.

About issue 7, I am not able to change the height of the native image picker for iOS, so for the time being I just hid the alert banner on this screen (only in iOS). We can further improve it later if needed, wdyt?

**Comment by pavloburykh** [comment]
> Thank you @pavloburykh for finding issue 6, it was an interesting one. Should be fixed now.

Hey @Parveshdhull! Thanks a lot for fixing the issue. Looks good to me now!

> About issue 7, I am not able to change the height of the native image picker for iOS, so for the time being I just hid the alert banner on this screen (only in iOS). We can further improve it later if needed, wdyt?

I think hiding banner on this screen is a good a solution and we are fine to go with it, thank you.

@Parveshdhull I have noticed another issue. Please take a look. I believe I can log it as followup if you prefer to get this PR merged first. 

### ISSUE 8 Carousel with image preview is cut in gallery view mode

Affects both IOS and Android

**Steps:** 

1. Send a message with image
2. Open the sent image
3. Pay attention at carousel with image preview  at the bottom. 

**Actual result:** image preview is cut. 


https://github.com/status-im/status-mobile/assets/97245802/0d4ce40d-8f79-45e8-b029-383b7e73439f

**Android:**

photo_2024-04-29 15 00 20

**IOS:** 

photo_2024-04-29 15 00 24

**Comment by flexsurfer** [comment]
it doesn't look that it's worth it, should we just simplify the banner? cc @alwx

**Comment by pavloburykh** [comment]
> it doesn't look that it's worth it, should we just simplify the banner? cc @alwx

Hey @flexsurfer! 
@Parveshdhull has already done a great amount of job and successfully fixed most of banner related issues.

**Comment by Parveshdhull** [comment]
thank you @pavloburykh for finding issue 8, should be fixed now.

UPD: component-tests failed will push a fix.

**Comment by Parveshdhull** [comment]
upd2: failed component tests were not related to PR, rerunning worked

**Comment by Parveshdhull** [comment]
Thank you very much Pavlo for all your help and patience while testing the PR.
As discussed merging the PR with fixes. Thank you

**Pull request merged**